test/AdapterTests.cpp

```diff
@@ -278,6 +278,106 @@ TEST_CASE( "Test source mode", "[source]") {
     tcp_adapter_thread.join();
 }
 
+TEST_CASE( "Test source mode with client token", "[source]") {
```

Comment:
Any test of `validate_client_token`? If so, then Catch2 framework generators could help https://github.com/catchorg/Catch2/blob/devel/docs/generators.md

Reply:
I just removed it because I realized the service responds with:

```
[2022-08-11T18:03:22.100299]{7}[error]   Proxy server rejected web socket upgrade request: (HTTP/1.1 400 Bad Request) "client-token must match ^[a-zA-Z0-9-]{32,128}$"
```